### ISSUE TITLE
Allow the installation to complete even with SMTP port blocked

### ISSUE THREAD

**Issue by gmeans** [omega8cc/boa#1509]
I'm attempting to install on a fresh Debian Stretch VM on GCP, and am running into issues. I first tried the `in-stable` installation, but after reviewing issues in the repo I saw that `in-head` was recommended. I tried that as well with the same results.

`boa in-head public [server.mydomain.org] [my@email] [o1]` 
> Items in [] were replaced with my own values.

```
I don't have the exact message, but it implied octopus couldn't run because barracuda wasn't up to date with the latest version.
```

I rebooted the server, and then attempted to run the following as mentioned in the logs from the initial install.

`barracuda up-head system`

```
ERROR: I have no idea where PHP-CLI is!
```

`apt-get install php-cli`
`barracuda up-head system`

```
BOA [20:18:10] ==> Skynet Agent v.BOA-4.1.1-dev welcomes you aboard!

BOA [20:18:13] ==> INFO: NORMAL INIT
BOA [20:18:13] ==> FATAL ERROR: You must define also _EASY_HOSTNAME
```

In a previous pass I did attempt setting that, but it failed with the same error. 

I'd appreciate any ideas for me to try. Thanks!

**Comment by omega8cc** [comment]
There was a temporary issues caused by switching to Percona on new installs be default. We have reverted it back to MariaDB for now, and it works again, but unfortunately you will need to start from scratch, since this instance failed to install the main hostmaster instance silently and it's not possible to easily fix this other than by clean rebuild. Sorry for the confusion!

**Comment by gmeans** [comment]
Hi, just wanted to report back that this wasn't resolved. I reattempted twice (the first I was concerned maybe I did in-stable vs. in-head).

On the second round I opted to create the `.barracuda.cnf` file prior to installing and this time I noticed that PHP was built (both 5 and 7). I copied the template available in the repo.

However, once I attempted the `barracuda up-head system` I received the same error about `_EASY_HOSTNAME`. This value is set to the correct DNS entry.

```
ems-boa-1:~# tail -f /var/backups/reports/up/barracuda/200719/barracuda-up-200719-200802.log

BOA [20:08:26] ==> Skynet Agent v.BOA-4.1.1-dev welcomes you aboard!

BOA [20:08:29] ==> INFO: FORCED INIT
BOA [20:08:30] ==> FATAL ERROR: You must define also _EASY_HOSTNAME
```

I'm happy to send whatever you'd like for troubleshooting. Or if it's easier, enable you to ssh into the server.

Thanks!

**Comment by omega8cc** [comment]
Clean install works fine, please don't try any custom tricks, just install and then upgrade BOA head after reboot on a clean VPS.

**Comment by gmeans** [comment]
Thanks for all of your replies. I ended up resolving my issues by using Digital Ocean vs. GCP.

tl;dr: GCP is probably a no go for BOA due to blocking 25, Digital Ocean seems ok.

The reason I had tried the `.barracuda.cnf` was due to me noticing the SMTP check failed. I wanted to set that to NO to see if it helped. 

This morning I did an install side-by-side between a Digital Ocean VPS and a GCP VPS. **GCP blocks port 25 and you are unable to open it**. This causes to initial installation to fail before PHP is installed which prevents you from using upgrade and changing the `.barracuda.cnf` settings.

The error message when this happens says to define the relay host and re-enable the command. I attempted to set the `_SMTP_RELAY_TEST=No` and rerun the boa command, however it detects previous installation and won't let you proceed.

```
BOA [09:59:13] ==> INFO: Fix #2 for libs in Debian stretch

      Your outgoing SMTP port 25 doesn't work
      as expected, so your server can't send out
      any emails directly.

      It appears you need a SMTP relayhost added
      in your Postfix /etc/postfix/main.cf config.

      Please enter your SMTP relay host name
      as a value for _SMTP_RELAY_HOST="smtp.your.relay.server"

      Then run this script again.


BOA [09:59:38] ==> Skynet Agent v.BOA-4.1.1-dev welcomes you aboard!

BOA [09:59:41] ==> ERROR: This octopus installer can be used only when the same version
BOA [09:59:41] ==> of boa or barracuda installer was used before. Your system must be
BOA [09:59:41] ==> upgraded with barracuda installer version BOA-4.1.1-dev first
BOA [09:59:41] ==> Bye

BOA in-head completed
Bye
```

```
root@ems-boa-1:~# boa in-head public ems-boa-gcp.ceroic.com grant@ceroic.com ems
ERROR: BOA requires minimal supported OS with no services installed.

Please make sure you don't have MySQL nor Apache installed.
Here's the list of directories which shouldn't exist:
/etc/nginx /etc/apache2 /etc/mysql /var/lib/mysql /var/aegir
Bye
root@ems-boa-1:~#
```

My issue really stemmed from me not realizing the initial installed failed w/o any chance of recovery. This is the messaging I saw during this install:

```

      Your outgoing SMTP port 25 doesn't work
      as expected, so your server can't send out
      any emails directly.

      It appears you need a SMTP relayhost added
      in your Postfix /etc/postfix/main.cf config.

      Please enter your SMTP relay host name
      as a value for _SMTP_RELAY_HOST="smtp.your.relay.server"

      Then run this script again.


BOA [09:59:38] ==> Skynet Agent v.BOA-4.1.1-dev welcomes you aboard!

BOA [09:59:41] ==> ERROR: This octopus installer can be used only when the same version
BOA [09:59:41] ==> of boa or barracuda installer was used before. Your system must be
BOA [09:59:41] ==> upgraded with barracuda installer version BOA-4.1.1-dev first
BOA [09:59:41] ==> Bye

BOA in-head completed
Bye
```

It appears things completed and I have 2 options at this point. Add an _SMTP_RELAY_HOST and / or upgrade barracuda. But in reality the main installation failed and isn't at a point you can do anything.

Hope this helps improve what's really a fantastic hosting setup. Thanks again!

**Comment by omega8cc** [comment]
Thank you for the detailed feedback, it's always highly appreciated!

We could perhaps make it so the blocked port 25 results only with warning, to let you known that you should check the on-disk copy of welcome email, because it can't be sent while port 25 is blocked, but otherwise allow the installation to continue.
